File: dataset/human/c/file_7048.c
#include <stdio.h>
#include <math.h>

int main(void){

  int h, w;
  int i, j;

  for(;;){
    scanf("%d %d", &h, &w);

    if(h == 0 && w == 0) break;
  
    for(i = 0; i < h; i++){
      for(j = 0; j < w; j++){
	if(i % 2 == 0){
	  if(j % 2 == 0) printf("#");
	  else printf(".");
	}else{
	  if(j % 2 == 0) printf(".");
	  else printf("#");
	}
	if(j == w-1) printf("\n");
      }
      if(i == h-1) printf("\n");
    }
  }

  return 0;

}

File: dataset/human/java/file_7048.java
import java.util.Scanner;


	public class Main {
		
		public static void main(String[] args) {

			Scanner sc = new Scanner(System.in);
			
			long k = sc.nextLong();
			long n = sc.nextLong();
			long[] a = new long[(int)n];
			
			for(int i=0;i<n;i++) {
				a[i] = sc.nextLong();
			}
			long dis = 0;
			for(int i=1;i<n;i++) {
				dis = Math.max(dis, a[i]-a[i-1]);
			}
			
			dis = Math.max(dis, Math.abs(k-a[(int)n-1]+a[0]));
			
			System.out.println(k-dis);
			
	    }
		
	}
